web/app/view/ReportController.js

```diff
@@ -77,6 +90,152 @@ Ext.define('Traccar.view.ReportController', {
     },
 
     selectReport: function (position, center) {
-        this.getView().getSelectionModel().select([position], false, true);
+        if (position instanceof Traccar.model.Position) {
+            this.getView().getSelectionModel().select([position], false, true);
+        }
+    },
+
+    downloadCsv: function (requestUrl, requestParams) {
+        Ext.Ajax.request({
+            url: requestUrl,
+            method: 'GET',
+            params: requestParams,
+            headers: {
+                Accept: 'text/csv'
+            },
+            success: function (response) {
```

Comment:
It was the hardest part of code to pass header and force browser to download result.
`Ext.form.action.StandardSubmit` replaces header to his own, "nonstandart submit" do not force browser show "save as" dialog.

I can replace DOM manipulation to `Ext.DomHelper` but idea should stay the same